src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelBase.cs

```diff
@@ -246,6 +249,14 @@ protected ViewModelBase(bool supportIEditableObject, bool ignoreMultipleModelsWa
             DependencyResolver = serviceLocator.ResolveType<IDependencyResolver>();
             _dispatcherService = DependencyResolver.Resolve<IDispatcherService>();
 #endif
+            var type = GetType();
+
+            serviceLocator.RegisterTypeIfNotYetRegisteredWithTag(typeof(IObjectIdGenerator<int>), typeof(IntegerObjectIdGenerator<>).MakeGenericType(type), type);
```

Comment:
MakeGenericTypeEx, and we need to measure performance whether we can / should cache. The reason is that MakeGenericType will *always* be executed, even when already registered. Instead, we should separate into 2 calls to first check whether it's registered, *or* the RegisterTypeIfNotYetRegisteredWithTag should get an Func<> overload to lazy-create the type.

Reply:
Fixed